src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -501,7 +512,16 @@ def sql_query(self) -> str:
             if self.offset
             else analyzer_utils.EMPTY_STRING
         )
-        self._sql_query = f"{analyzer_utils.SELECT}{self.projection_in_str}{analyzer_utils.FROM}{from_clause}{where_clause}{order_by_clause}{limit_clause}{offset_clause}"
+        if exclude_clause != analyzer_utils.EMPTY_STRING:
+            self._sql_query = (
+                f"{analyzer_utils.SELECT}{analyzer_utils.STAR}{exclude_clause}{analyzer_utils.FROM}"
+                f"{from_clause}{where_clause}{order_by_clause}{limit_clause}{offset_clause}"
+            )
+        else:
+            self._sql_query = (
+                f"{analyzer_utils.SELECT}{self.projection_in_str}{exclude_clause}{analyzer_utils.FROM}"
+                f"{from_clause}{where_clause}{order_by_clause}{limit_clause}{offset_clause}"
+            )
```

Comment:
You could use `selected` or `projection` as a variable name, such that it represents either `{analyzer_utils.STAR}{exclude_clause}` or `{self.projection_in_str}` depending on the value of `exclude_clause. This way you don't need to replicate most of the other clauses of the query.

Also, should we assert that only one of `projection_in_str` and `exclude_clause` exists?

Reply:
The assertion that only one exists is implicit in the flattening checks; for example https://github.com/snowflakedb/snowpark-python/pull/949/files#diff-ba9463440d935e1d5667a2909f7d2b8061f56de7bdabd8207a29903e9eae8f59R681-R682.

Having an extra assertion might help though, I will add that, thank you!